core/src/main/java/org/jdbi/v3/core/locator/ClasspathSqlLocator.java

```diff
@@ -75,7 +79,11 @@ private static String resourcePathFor(String fullyQualifiedName) {
     }
 
     private static String findSqlOnClasspath(ClassLoader classLoader, String path) {
-        return CACHE.computeIfAbsent(path, p -> readResource(classLoader, p));
+        try {
+            return CACHE.get(path, () -> readResource(classLoader, path));
```

Comment:
Just provide the loader function to the `CacheBuilder.build` method as a lambda expression. Moves the lambda from here to the cache declaration